jwst/resample/resample.py

```diff
@@ -1,357 +1,585 @@
-                xmin=xmin,
-                xmax=xmax,
-                ymin=ymin,
-                ymax=ymax,
-            )
-            del data
-
-            # make an approximate error image by drizzling it
-            # in the same way the image is handled
-            if compute_error:
-                driz_error.add_image(
-                    data=img.err,
-                    exptime=img.meta.exposure.exposure_time,  # GWCSDrizzle.add_image param default
-                    pixmap=pixmap,
-                    scale=iscale,
-                    weight_map=inwht,
-                    wht_scale=1.0,  # hard-coded for JWST count-rate data
-                    pixfrac=self.pixfrac,
-                    in_units="cps",  # GWCSDrizzle.add_image param default
-                    xmin=xmin,
-                    xmax=xmax,
-                    ymin=ymin,
-                    ymax=ymax,
-                )
-            input_models.shelve(img, index, modify=False)
-            del img
+        """
+        if self.output_jwst_model is not None:
+            self.reset_arrays(reset_output=True)
 
-        output_model.data = driz.out_img
-        output_model.wht = driz.out_wht
-        del driz
-        # copy the drizzled error into the output model
-        if compute_error:
-            output_model.err = driz_error.out_img
-            del driz_error
+        output_model_filename = ''
 
-        return output_model
+        log.info(f"{len(indices)} exposures to drizzle together")
+        first = True
+        with self.input_models:
+            for index in indices:
+                model = self.input_models.borrow(index)
+                if self.output_jwst_model is None:
+                    # Determine output file type from input exposure filenames
+                    # Use this for defining the output filename
+                    indx = model.meta.filename.rfind('.')
+                    output_type = model.meta.filename[indx:]
+                    output_root = '_'.join(model.meta.filename.replace(
+                        output_type,
+                        ''
+                    ).split('_')[:-1])
+                    output_model_filename = (
+                        f'{output_root}_'
+                        f'{self.intermediate_suffix}{output_type}'
+                    )
 
-    def resample_many_to_many(self, input_models):
+                if isinstance(model, datamodels.SlitModel):
+                    # must call this explicitly to populate area extension
+                    # although the existence of this extension may not be
+                    # necessary
+                    model.area = model.area
+
+                self.add_model(model)
+                if first:
+                    self.output_jwst_model.meta.bunit_data = model.meta.bunit_data
+                    first = False
+                self.input_models.shelve(model, index, modify=False)
```

Comment:
```suggestion
                self.input_models.shelve(model, index)
```
The model is modified on line 568.